keyvaluestore/nodewriter.go

```diff
@@ -17,7 +17,38 @@ func (writer *NodeWriterImpl) WriteNode(node *node) error {
 		return nil
 	}
 
-	//id := node.nodeId
+	id := node.nodeId
 
-	return nil
+	bytes := ConvertBool(node.isLeaf)
+	bytes = append(bytes, ConvertUInt64(id)...)
+	bytes = append(bytes, ConvertUInt32(node.n)...)
+	bytes = append(bytes, ConvertUInt64(node.next)...)
+	bytes = append(bytes, ConvertUInt64(node.parent)...)
+
+	var i uint32 = 0
+	for ; i < node.n; i++ {
+		bytes = append(bytes, ConvertUInt64(node.keys[i])...)
+	}
+
+	if node.isLeaf {
+		i = 0
+		for ; i < node.n; i++ {
+			bytes = append(bytes, node.values[i][:]...)
+		}
+	} else {
+		i = 0
+		for ; i < node.n+1; i++ {
+			bytes = append(bytes, ConvertUInt64(node.children[i])...)
+		}
+	}
+
+	// fill rest of page with 0s
```

Comment:
Not really necessary, no?

Reply:
If a node gets split, the size might decrease. It is not 100% necessary, because we only read the first n (or n+1) keys, values and children. I still think its a good idea to remove "dead" data from the file.

Reply:
If a node gets split, the size might decrease. It is not 100% necessary, because we only read the first n (or n+1) keys, values and children. I still think its a good idea to remove "dead" data from the file.